src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,230 @@
+            state[2] = 0x98badcfe;
+            state[3] = 0x10325476;
+            // Zeroize sensitive information
+            buffer.Fill(0);
+
+            HashCore(source, state, count, buffer);
+
+            /* Save number of bits */
+            Span<byte> bits = stackalloc byte[8];
+            Encode(bits, count);
+
+            /* Pad out to 56 mod 64. */
+            uint index = ((count[0] >> 3) & 0x3f);
+            int padLen = (int)((index < 56) ? (56 - index) : (120 - index));
+            Span<byte> padding = stackalloc byte[padLen];
+            padding[0] = 0x80;
+            HashCore(padding, state, count, buffer);
+
+            /* Append length (before padding) */
+            HashCore(bits, state, count, buffer);
+
+            /* Write state to destination */
+            Encode(destination, state);
+        }
+
+        private static void HashCore(ReadOnlySpan<byte> input, Span<uint> state, Span<uint> count, Span<byte> buffer)
+        {
+            /* Compute number of bytes mod 64 */
+            int index = (int)((count[0] >> 3) & 0x3F);
+            /* Update number of bits */
+            count[0] += (uint)(input.Length << 3);
+            if (count[0] < (input.Length << 3))
+                count[1]++;
+            count[1] += (uint)(input.Length >> 29);
+
+            int partLen = 64 - index;
+            int i = 0;
+            /* Transform as many times as possible. */
+            if (input.Length >= partLen)
+            {
+                BlockCopy(input, 0, buffer, index, partLen);
+                MD4Transform(state, buffer);
+
+                for (i = partLen; i + 63 < input.Length; i += 64)
+                {
+                    MD4Transform(state, input.Slice(i));
+                }
+
+                index = 0;
+            }
+
+            /* Buffer remaining input */
+            input.Slice(i).CopyTo(buffer.Slice(index));
+        }
+
+        //--- private methods ---------------------------------------------------
+
+        private static void BlockCopy(ReadOnlySpan<byte> source, int srcOffset, Span<byte> destination, int dstOffset, int count)
+        {
+            for (int srcIndex = srcOffset, dstIndex = dstOffset; srcIndex < srcOffset + count; srcIndex++, dstIndex++)
+            {
+                destination[dstIndex] = source[srcIndex];
+            }
+        }
+
+        /* F, G and H are basic MD4 functions. */
+        private static uint F(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((~x) & (z)));
+        }
+
+        private static uint G(uint x, uint y, uint z)
+        {
+            return (uint)(((x) & (y)) | ((x) & (z)) | ((y) & (z)));
+        }
+
+        private static uint H(uint x, uint y, uint z)
+        {
+            return (uint)((x) ^ (y) ^ (z));
+        }
+
+        /* FF, GG and HH are transformations for rounds 1, 2 and 3 */
+        /* Rotation is separate from addition to prevent recomputation */
```

Comment:
I am not sure what this comment means. The code is doing what the RFC says, ie `a = (a + G(b,c,d) + X[k] + 5A827999) <<< s` .. not sure where one could do recomputation.

Reply:
This comment seem to come from the original rfc1320 implementation, which was written in C.  (https://datatracker.ietf.org/doc/html/rfc1320 , page9).